src/contexts/BuySell/BuySellProvider.tsx

```diff
@@ -58,26 +61,40 @@ const BuySellProvider: React.FC = ({ children }) => {
     setSelectedCurrency(currencyTokens[0])
   }, [])
 
-  let spendingTokenBalance = new BigNumber(0)
-  if (!isUserBuying && buySellToken === 'index') {
-    spendingTokenBalance = indexBalance || new BigNumber(0)
-  } else if (!isUserBuying && buySellToken === 'dpi') {
-    spendingTokenBalance = dpiBalance || new BigNumber(0)
-  } else if (!isUserBuying && buySellToken === 'ethfli') {
-    spendingTokenBalance = ethfliBalance || new BigNumber(0)
-  } else if (!isUserBuying && buySellToken === 'btcfli') {
-    spendingTokenBalance = btcfliBalance || new BigNumber(0)
-  } else if (!isUserBuying && buySellToken === 'cgi') {
-    spendingTokenBalance = cgiBalance || new BigNumber(0)
-  } else if (!isUserBuying && buySellToken === 'mvi') {
-    spendingTokenBalance = mviBalance || new BigNumber(0)
-  } else if (selectedCurrency?.id === 'wrapped_eth') {
-    spendingTokenBalance = ethBalance || new BigNumber(0)
-  } else if (selectedCurrency?.id === 'mcd') {
-    spendingTokenBalance = daiBalance || new BigNumber(0)
-  } else if (selectedCurrency?.id === 'usdc') {
-    spendingTokenBalance = usdcBalance || new BigNumber(0)
-  }
+  useMemo(() => {
+    if (!isUserBuying && buySellToken === 'index') {
+      setSpendingTokenBalance(indexBalance || new BigNumber(0))
+    } else if (!isUserBuying && buySellToken === 'dpi') {
+      setSpendingTokenBalance(dpiBalance || new BigNumber(0))
+    } else if (!isUserBuying && buySellToken === 'ethfli') {
+      setSpendingTokenBalance(ethfliBalance || new BigNumber(0))
+    } else if (!isUserBuying && buySellToken === 'btcfli') {
+      setSpendingTokenBalance(btcfliBalance || new BigNumber(0))
+    } else if (!isUserBuying && buySellToken === 'cgi') {
+      setSpendingTokenBalance(cgiBalance || new BigNumber(0))
+    } else if (!isUserBuying && buySellToken === 'mvi') {
+      setSpendingTokenBalance(mviBalance || new BigNumber(0))
+    } else if (selectedCurrency?.id === 'wrapped_eth') {
+      setSpendingTokenBalance(ethBalance || new BigNumber(0))
+    } else if (selectedCurrency?.id === 'mcd') {
+      setSpendingTokenBalance(daiBalance || new BigNumber(0))
+    } else if (selectedCurrency?.id === 'usdc') {
+      setSpendingTokenBalance(usdcBalance || new BigNumber(0))
+    }
+  }, [
+    isUserBuying,
+    buySellToken,
+    indexBalance,
+    dpiBalance,
+    ethfliBalance,
+    btcfliBalance,
+    cgiBalance,
+    mviBalance,
+    ethBalance,
+    daiBalance,
+    usdcBalance,
+    selectedCurrency?.id,
+  ])
```

Comment:
It's standard practice to put something like this into a memo since the IO is fixed, in a way, and post-dependabot this was being flagged in a warning